extensions-core/hdfs-storage/src/main/java/io/druid/storage/hdfs/tasklog/HdfsTaskLogs.java

```diff
@@ -123,6 +125,28 @@ public void killAll() throws IOException
     FileSystem fs = taskLogDir.getFileSystem(hadoopConfig);
     fs.delete(taskLogDir, true);
   }
+
+  @Override
+  public void kill(long beforeTimestamp) throws IOException
+  {
+    Path taskLogDir = new Path(config.getDirectory());
+    FileSystem fs = taskLogDir.getFileSystem(hadoopConfig);
+    if (fs.exists(taskLogDir)) {
+      RemoteIterator<LocatedFileStatus> iter = fs.listLocatedStatus(taskLogDir);
+      while (iter.hasNext()) {
```

Comment:
There's no failure recovery in this loop, and I'm forgetting how HDFS handles various issues, is this intended to just fail this kill run if there are any sort of issues?